### PR TITLE
[Dotnet] port of red green notifications

### PR DESCRIPTION
**Pull request by stephen-hawley**

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: dotnet/ios/ios12/RedGreenNotifications/RedGreenNotifications/AppDelegate.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                        Console.WriteLine ($"ACTION: {response.ActionIdentifier}");
+                }
+
+                completionHandler ();
+        }
+
+        [Export ("userNotificationCenter:willPresentNotification:withCompletionHandler:")]
+        public void WillPresentNotification (UNUserNotificationCenter center, UNNotification notification, System.Action<UNNotificationPresentationOptions> completionHandler)
+        {
+                completionHandler (UNNotificationPresentationOptions.Alert | UNNotificationPresentationOptions.Sound);
```
Comment (tj-devel709) [inline dotnet/ios/ios12/RedGreenNotifications/RedGreenNotifications/AppDelegate.cs:74]: Same thing with UNNotificationPresentationOptions.Alert as above

File: dotnet/ios/ios12/RedGreenNotifications/RedGreenNotifications/AppDelegate.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        {
+                get;
+                set;
+        }
+
+        public override bool FinishedLaunching (UIApplication application, NSDictionary launchOptions)
+        {
+                // Request authorization to send notifications
+                UNUserNotificationCenter center = UNUserNotificationCenter.Current;
+                var options = UNAuthorizationOptions.ProvidesAppNotificationSettings | UNAuthorizationOptions.Alert | UNAuthorizationOptions.Sound | UNAuthorizationOptions.Provisional;
```
Comment (tj-devel709) [inline dotnet/ios/ios12/RedGreenNotifications/RedGreenNotifications/AppDelegate.cs:20]: This UNNotificationPresentationOptionAlert is deprecated in iOS 14: https://developer.apple.com/documentation/usernotifications/unnotificationpresentationoptions/unnotificationpresentationoptionalert?language=objc

Here's what I did when running into this: https://github.com/xamarin/ios-samples/pull/430/files#diff-ef95eabaf51b399f1dd83f7d31bd9deac783634e11ed44b8f4084d6aa33f6e8fR27

**Review by tj-devel709 — COMMENTED** [review]
Other than UNNotificationPresentationOptions.Alert being deprecated, everything else looks good to me!

File: dotnet/ios/ios12/RedGreenNotifications/RedGreenNotifications/AppDelegate.cs (hunk 2 above)
Comment (stephen-hawley) [reply dotnet/ios/ios12/RedGreenNotifications/RedGreenNotifications/AppDelegate.cs:20]: This one is a different enum. Alert is not deprecated here.

**Pull request merged**